File: Patrones Comportamiento/Strategy/Strategy.py
# Strategy
class Strategy:
    def execute(self, a, b): pass

class Multiply(Strategy):
    def execute(self, a, b):
        return a * b

class Context:
    def __init__(self, strategy):
        self.strategy = strategy

    def run(self, a, b):
        return self.strategy.execute(a, b)

print(Context(Multiply()).run(3, 4))

File: Patrones Comportamiento/Strategy/Strategy.js
// Strategy
class Strategy {
  execute(a, b) {}
}

class Multiply extends Strategy {
  execute(a, b) {
    return a * b;
  }
}

class Context {
  constructor(strategy) {
    this.strategy = strategy;
  }
  run(a, b) {
    return this.strategy.execute(a, b);
  }
}

console.log(new Context(new Multiply()).run(3, 4));
